plugins/source/azure/resources/services/monitor/diagnostic_settings.go

```diff
@@ -0,0 +1,80 @@
+package monitor
+
+import (
+	"context"
+	"errors"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+type diagnosticSettingsWrapper struct {
+	*armmonitor.DiagnosticSettingsResource
+	ResourceId string
+}
+
+func DiagnosticSettings() *schema.Table {
+	return &schema.Table{
+		Name:        "azure_monitor_diagnostic_settings",
+		Resolver:    fetchDiagnosticSettings,
+		Description: "https://learn.microsoft.com/en-us/rest/api/monitor/diagnostic-settings/list?tabs=HTTP#diagnosticsettingsresource",
+		Multiplex:   client.SubscriptionMultiplexRegisteredNamespace("azure_monitor_diagnostic_settings", client.Namespacemicrosoft_insights),
+		Transform:   transformers.TransformWithStruct(&diagnosticSettingsWrapper{}, transformers.WithPrimaryKeys("ID")),
```

Comment:
```suggestion
		Transform:   transformers.TransformWithStruct(&diagnosticSettingsWrapper{}, transformers.WithPrimaryKeys("ID"), transformers.WithUnwrapStructFields("DiagnosticSettingsResource")),
```

Since we're wrapping the original struct, you'll need to unwrap it to get the properties